package/MDAnalysis/coordinates/PDB.py

```diff
@@ -600,23 +608,24 @@ def _read_frame(self, frame):
             for line in f:
                 if line[:6] == 'ENDMDL':
                     break
-                # NOTE - CRYST1 line won't be found if it comes before the MODEL
-                # line, which is sometimes the case, e.g. output from gromacs
-                # trjconv
+                # NOTE - CRYST1 line won't be found if it comes before the
+                # MODEL line, which is sometimes the case, e.g. output from
+                # gromacs trjconv
                 elif line[:6] == 'CRYST1':
-                    A, B, C = map(float, [line[6:15], line[15:24], line[24:33]])
-                    alpha, beta, gamma = map(float, [line[33:40], line[40:47],
+                    A, B, C = np.float32([line[6:15], line[15:24],
+                                          line[24:33]])
+                    alpha, beta, gamma = np.float32([line[33:40], line[40:47],
                                                      line[47:54]])
                     self.ts._unitcell[:] = A, B, C, alpha, beta, gamma
                     continue
                 elif line[:6] in ('ATOM  ', 'HETATM'):
                     # we only care about coordinates
-                    self.ts._pos[pos] = list(map(float, [line[30:38],
-                                                         line[38:46],
-                                                         line[46:54]]))
+                    self.ts._pos[pos] = np.float32([line[30:38],
```

Comment:
This is not performance critical so readability counts. The simplest (and fastest) solution is just
```python
self.ts._unitcell[:] = line[6:15], line[15:24], line[24:33],  # A, B, C
                              line[33:40], line[40:47],  line[47:54]  # alpha, beta, gamma
```
(see below for quick benchmarks).

More importantly, for the *coordinates* you can use the same trick and that might actually give you a real additional speed up of a factor of 2!


```python
In[14]: dims = np.zeros(6, dtype=np.float32)
In [15]: %timeit dims[:3] = np.float32(["1", "2", ".3"])
The slowest run took 10.94 times longer than the fastest. This could mean that an intermediate result is being cached
100000 loops, best of 3: 4.21 µs per loop

In [16]: %timeit dims[:3] = map(float, ["1", "2", ".3"])
The slowest run took 5.29 times longer than the fastest. This could mean that an intermediate result is being cached
100000 loops, best of 3: 3.97 µs per loop

In [17]: %timeit dims[:3] = ["1", "2", ".3"]
The slowest run took 10.71 times longer than the fastest. This could mean that an intermediate result is being cached
1000000 loops, best of 3: 1.78 µs per loop

In [18]: %timeit dims[:3] = "1", "2", ".3"
The slowest run took 9.97 times longer than the fastest. This could mean that an intermediate result is being cached
1000000 loops, best of 3: 1.91 µs per loop
```